### PR TITLE
Split moves with seven

### LINKED ISSUE #105: On playing card 7, the user can either move 7 spaces forward with one of its marblesor can split it up and move in total 7 spaces with different marbles from it and his partner

### LINKED ISSUE #139: On playing card SEVEN, all marbles that the ball(s) travel(s) over get sent back to base, even marbles of his own color

### LINKED ISSUE #141: Backend: On joining a game, I want to be added to a team so that the player opposite of me is in the same team as me

### PR DESCRIPTION
**Pull request by SidlerD**
closes #105 
closes #139 
closes #141

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/GameLogicService.java [hunk 1]
```diff
@@ -40,21 +51,16 @@ else if (cardRank.equals(Rank.FOUR)) {
...
-        // TODO: implement cooperation with team member
-        // and killing on the way
-        // probably just a special number for the move
-        // which then will be processed as for loop of 1 distance moves
+       
         else if (cardRank.equals(Rank.SEVEN)) {
+            int holesTraveled = game.getHolesTravelled();
             // special move num
+            // FIXME: Do we actually need this?
             possibleMoves.add(107);
```
Comment (SidlerD) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/service/GameLogicService.java:59]: @shmnrr Do we still need this? I think you added this as special move num, but I handled it differently.

Comment (shmnrr) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/GameLogicService.java:59]: Yeah it was just a placeholder

**Comment by shmnrr** [comment]
image
Whenever trying to move with 7 it also highlights the current ball position - 21
When playing a seven in multiple moves, after each one it defaults to first card in hand, so you gotta click over and over on 7, but that's frontend.

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/GameLogicService.java (hunk 1 above)
Comment (SidlerD) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/GameLogicService.java:59]: Okay I'll remove it then

**Comment by SidlerD** [comment]
> image Whenever trying to move with 7 it also highlights the current ball position - 21 When playing a seven in multiple moves, after each one it defaults to first card in hand, so you gotta click over and over on 7, but that's frontend.

Yeah forgot to tell you but there's a branch on the frontend that fixes this (feat/move-with-seven). And the highlighting of the position -21 is due to the special move num that I wasn't sure if I could remove. Should all be fixed now

**Review by shmnrr — APPROVED** [review]

**Pull request merged**